task-sdk/src/airflow/sdk/bases/notifier.py

```diff
@@ -33,11 +32,31 @@
 
 
 class BaseNotifier(LoggingMixin, Templater):
-    """BaseNotifier class for sending notifications."""
+    """
+    BaseNotifier class for sending notifications.
+
+    It can be used asynchronously (preferred) if `async_notify`is implemented and/or
+    synchronously if `notify` is implemented.
+
+    Currently, the DAG/Task state change callbacks run on the DAG Processor and only support sync usage.
+
+    Usage:
```

Comment:
```suggestion
    Usage::
    
```

(Then it'll get rendered as a code block if these docs ever rendered by sphinx)

Reply:
I didn't realize that would work.  Neat.  I'll make the change and et back to you on the rest ASAP.  Thanks for the review.